### PR TITLE
API Contract for Skills APIs

### LINKED ISSUE #70: [Members Skills] API Contract
- Should have a Skills collection which will have 4 fields
    - Name
    - Awarded on
    - Awarded by
    - Why it was awarded
- Each user object will have a `Skills` property, which will be an array and contain the skills document id

### PR DESCRIPTION
**Pull request by Mir-SA**
Closes #70

### CODE REVIEW

File: skills/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Skills
+
+## Skills Object
+
+```
+{
+    "name": <name of skill>,
+    "awarded_on": <when skill was added>,
+    "awarded_by": <who added it>,
+    "awarded_for": <reason for adding it>
```
Comment (ankushdharkar) [inline skills/README.md:10]: Refer: https://github.com/Real-Dev-Squad/website-data-models/pull/7#discussion_r938592542

File: skills/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    "awarded_for": <reason for adding it> 
+}
+```
+
+## **Requests**
+
+|                     Route                      |         Description          |
+| :--------------------------------------------: | :--------------------------: |
+|           [GET /skills](#get-skills)           | Returns skills in collection |
+| [POST /skills/:username](#post-skillsusername) |  Adds skills to collection   |
```
Comment (ankushdharkar) [inline skills/README.md:19]: How do we GET the skills of a particular member by their username?

File: users/README.md [hunk 3]
```diff
@@ -18,6 +18,7 @@
   'linkedin_id': string,
   'twitter_id': string,
   'instagram_id': string,
+  'skills': [],
```
Comment (ankushdharkar) [inline users/README.md:21]: Refer: https://github.com/Real-Dev-Squad/website-data-models/pull/7#discussion_r938594653

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: skills/README.md (hunk 2 above)
Comment (Mir-SA) [reply skills/README.md:19]: we can fetch the `skills` of a user and query DB with `/skills/skillId` and return an array of their skills ...

File: skills/README.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+## **Requests**
+
+|                       Route                       |         Description          |
+| :-----------------------------------------------: | :--------------------------: |
+|            [GET /skills](#get-skills)             | Returns skills in collection |
+|  [POST /skills/:username](#post-skillsusername)   |  Adds skills to collection   |
+|   [GET /skills/:username](#get-skillsusername)    |   Get skills of given user   |
+| [GET /skills/users/:skill](#get-skillsusersskill) |  Get users with given skill  |
```
Comment (rohan09-raj) [inline skills/README.md:21]: This should be moved to `users` as it belongs to `users` and the route can be `/users/:skill`

File: skills/README.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  - **Code:** 200
+    - **Content:**
+```
+{
+  message: 'Skills returned successfully!'
+  tasks: [
+           {<skill_object>},
+           {<skill_object>}
+         ]
+}
```
Comment (rohan09-raj) [inline skills/README.md:44]: Can we please have JSON formatting here ?

File: skills/README.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+- **Success Response:**
+  - **Code:** 200
+    - **Content:**
+```
+{
+    message: "Skills returned successfully!"
+    data: [
+       "<skill_string>",
+       "<skill_string>",
+      ]
```
Comment (rohan09-raj) [inline skills/README.md:99]: Can we please have correct indentation ?

File: skills/README.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+- **Success Response:**
+  - **Code:** 200
+    - **Content:**
+```
+{
+  message: 'Users returned successfully!'
+  data: [
+           { <user_object> },
+           { <user_object> },
+         ]
```
Comment (rohan09-raj) [inline skills/README.md:125]: Can we please have correct indentation here ?

File: skills/README.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
+# Skills
+
+## Skills Object
+
+```
+{
+    "name": <name of skill>,
+    "on": <when skill was added>,
```
Comment (harshith-venkatesh) [inline skills/README.md:8]: createdOn/ createdAt would be better right?
Are we not thinking about levels here?

File: skills/README.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
+# Skills
+
+## Skills Object
+
+```
+{
+    "name": <name of skill>,
+    "on": <when skill was added>,
+    "by": <who added it>,
+    "for": <reason for adding it>
```
Comment (harshith-venkatesh) [inline skills/README.md:10]: Wouldn't reason sounds better than for here?

File: skills/README.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
+# Skills
+
+## Skills Object
+
+```
+{
+    "name": <name of skill>,
+    "on": <when skill was added>,
+    "by": <who added it>,
```
Comment (harshith-venkatesh) [inline skills/README.md:9]: createdBy/ Reviewer 
The purpose here is to ensure that the key signifies the meaning as much as possible

**Review by harshith-venkatesh — CHANGES_REQUESTED** [review]

File: skills/README.md (hunk 8 above)
Comment (Mir-SA) [reply skills/README.md:8]: 1. yes descriptive name would be better & i had mentioned on & by as `awarded_on` & `awarded_by` but ankush had suggested to just keep it short

Comment (Mir-SA) [reply skills/README.md:8]: 2. i think not. it was not mentioned by sumit in the referred ticket

File: skills/README.md (hunk 10 above)
Comment (Mir-SA) [reply skills/README.md:9]: https://github.com/Real-Dev-Squad/website-api-contracts/pull/86#discussion_r962937443

File: skills/README.md (hunk 8 above)
Comment (harshith-venkatesh) [reply skills/README.md:8]: createdAt is an standard term used in many projects that I am aware of.....
I am curious on why @ankushdharkar agreed with on and by.
However, if we have come to conclusion after discussion. We are cool

**Comment by prakashchoudhary07** [comment]
Closing this as this feature is being built as a stand-alone product, in Skill-tree

https://github.com/Real-Dev-Squad/skill-tree-frontend/
https://github.com/Real-Dev-Squad/skill-tree-backend/

**Pull request closed without merging**